dotCMS/src/main/java/org/apache/felix/framework/OSGIUtil.java

```diff
@@ -76,28 +76,28 @@
 
     public static OSGIUtil getInstance() {
         if (instance == null) {
-            instance = new OSGIUtil();
+            instance = new OSGIUtil(Config.CONTEXT);
```

Comment:
this is not thread safe

Reply:
meaning, we need something like a sync block and a double null check here?

Reply:
sync block and a double null check are bad for performance

I think the Bill Pugh Singleton is better

https://www.harinathk.com/java/singleton-bill-pugh-solution-enum/